tests/units/classes/Form/QgisFormControlPropertiesTest.php

```diff
@@ -0,0 +1,224 @@
+    }
+
+    function testGetValueRelationData() {
+
+        $properties = new QgisFormControlProperties(
+            'tram_id',
+            'ValueRelation',
+            'menulist',
+            array(
+                'AllowMulti' => false,
+                'AllowNull' => true,
+                'FilterExpression' => '',
+                'Key' => 'osm_id',
+                'Layer' => 'tramway20150328114206278',
+                'OrderByValue' => true,
+                'UseCompleter' => false,
+                'Value' => 'test',
+            )
+        );
+
+        $valueRelationData = $properties->getValueRelationData();
+        $this->assertTrue(is_array($valueRelationData));
+        $this->assertTrue($valueRelationData['allowNull']);
+        $this->assertTrue($valueRelationData['orderByValue']);
+        $this->assertEquals($valueRelationData['layer'], 'tramway20150328114206278');
+        $this->assertEquals($valueRelationData['layerName'], '');
+        $this->assertEquals($valueRelationData['key'], 'osm_id');
+        $this->assertEquals($valueRelationData['value'], 'test');
+        $this->assertFalse($valueRelationData['allowMulti']);
+        $this->assertEquals($valueRelationData['filterExpression'], '');
+        $this->assertFalse($valueRelationData['useCompleter']);
+        $this->assertTrue($valueRelationData['fieldEditable']);
+    }
+
+    function testGetRelationReference() {
+
+        $properties = new QgisFormControlProperties(
+            'risque',
+            'RelationReference',
+            'menulist',
+            array(
+                'AllowNull' => true,
+                'OrderByValue' => false,
+                'Relation' => 'tab_demand_risque_risque_66c_risque',
+                'MapIdentification' => false,
+                'ReferencedLayerName' => 'risque',
+                'ReferencedLayerId' => 'risque_66cb8d43_86b7_4583_9217_f7ead54463c3',
+            )
+        );
+
+        $relationReferenceData = $properties->getRelationReference();
+        $this->assertTrue(is_array($relationReferenceData));
+        $this->assertTrue($relationReferenceData['allowNull']);
+        $this->assertFalse($relationReferenceData['orderByValue']);
+        $this->assertEquals($relationReferenceData['relation'], 'tab_demand_risque_risque_66c_risque');
+        $this->assertFalse($relationReferenceData['mapIdentification']);
+        $this->assertTrue(is_array($relationReferenceData['filters']));
+        $this->assertCount(0, $relationReferenceData['filters']);
+        $this->assertEquals($relationReferenceData['filterExpression'], Null);
+        $this->assertFalse($relationReferenceData['chainFilters']);
+        $this->assertEquals($relationReferenceData['referencedLayerName'], 'risque');
+        $this->assertEquals($relationReferenceData['referencedLayerId'], 'risque_66cb8d43_86b7_4583_9217_f7ead54463c3');
+
+        # AllowNULL / not AllowNull
```

Comment:
Can you add a note which QGIS version affected this code change ?

Reply:
@Gustry I don't know exactly which version introduces this change and my code prevents futur upper/lower changes.

Reply:
I thought you said 3.22.7 on the chat ... Ok anyway